compiler/damlc/daml-stdlib-src/DA/Internal/Exception.daml

```diff
@@ -119,6 +121,28 @@ instance HasFromAnyException ArithmeticError where
 
 --------------------------------------------------------------
 
+instance HasField "message" ContractError Text where
+    getField (ContractError m) = m
+    setField m (ContractError _) = ContractError m
+
+instance HasThrow ContractError where
+    throwPure = primitive @"EThrow"
+
+instance HasMessage ContractError where
+    message (ContractError m) = m
+
+instance HasToAnyException ContractError where
+    toAnyException = primitive @"EToAnyException"
+
+instance HasFromAnyException ContractError where
+    fromAnyException e
+        | primitive @"BEAnyExceptionIsContractError" e =
+            Some (ContractError (message e))
+        | otherwise =
+            primitive @"EFromAnyException" e
```

Comment:
Since we don't want the primitive any more, this will just become:

```Haskell
fromAnyException = primitive @"EFromAnyException"
```

And that's all that needs to change in this file. Then we can remove the primitive everywhere from the Haskell side as well.